src/rules.rs

```diff
@@ -303,6 +312,30 @@ impl ir::StructDatum {
 
             clauses.push(is_local);
         }
+        // If a type is `extern`, but is also `#[fundamental]`, it satisfies IsLocalTy
+        // if and only if its parameters satisfy IsLocalTy
+        else if self.binders.value.flags.fundamental {
```

Comment:
Nit: formatting wise, i'd prefer the comment to be inside the `else if` and to have `} else if .. {` on one line. This is because, when I am reading code, if the `else` is not on the same line as the closing block, I tend to assume there isn't one (I've noticed this pattern before...)